pkg/data/queue.go

```diff
@@ -0,0 +1,72 @@
+package data
+
+import (
+	"fmt"
+)
+
+// ErrorPlayerNotFound : Player specific errors
+var ErrorPlayerNotFound = fmt.Errorf("Player not found")
+
+// Player defines the structure for an player in queue.
+// Formatting done with json tags to the right. "-" : don't include when encoding to json
+type Player struct {
+	UserID          int     `json:"userid" validate:"required,notinqueue,exist"`
+}
+
+// Players is a collection of Player
+type Players []*Player
+
+
+// GetQueue returns the list of players in queue
+func GetQueue() Players {
+	return queue
+}
+
+// InQueue returns a boolean that verifies that a player is queued
+func InQueue(id int) bool {
+	return findIndexByPlayerID(id) > 0
+}
+
+// AddPlayer append a player to the queue
+func AddPlayer(player *Player) {
```

Comment:
There should be a condition here that pops the queue when 4 players are in queue. We then send a POST request to the dispatcher with the information on the players in the game. The condition and the post request should be implemented either here or in the handler itself.

Reply:
Probably here since we want the business logic to not be in the handlers